crates/rspack_plugin_javascript/src/plugin/side_effects_flag_plugin.rs

```diff
@@ -413,16 +417,12 @@ pub struct SideEffectsFlagPlugin;
 #[async_trait]
 impl Plugin for SideEffectsFlagPlugin {
   async fn optimize_dependencies(&self, compilation: &mut Compilation) -> Result<Option<()>> {
-    // SAFETY: this method will not modify the map, and we can guarantee there is no other
-    // thread access the map at the same time.
+    let entries = compilation.entry_modules().collect::<Vec<_>>();
     let mg = &mut compilation.module_graph;
```

Comment:
We constructed module graph parallel, but the inserted module is out of order, use level order to align with webpack.